### ISSUE TITLE
OSD pods cannot recover after kubelet restart: cephosd: skipping device <device> that is in use (not by rook). fs: , ownPartitions: false

### ISSUE THREAD

**Issue by ivanovaleksandar** [rook/rook#1880]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When restarting a kubelet/node (data nodes) that had osd pods running on it, it seems that the OSDs are does and cannot recover, so as I result, the PVC/PV cannot be attached to the desired pods.

Here are some logs and execs from commands that should help with troubleshooting:
```
kubectl logs rook-agent-dm4qx

2018-07-13 08:11:06.247313 I | flexdriver: calling agent to attach volume rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864
2018-07-13 08:11:06.250122 I | ceph-volumeattacher: attaching volume rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864 cluster rook
2018-07-13 08:11:06.253710 I | cephmon: parsing mon endpoints: rook-ceph-mon4=10.106.8.20:6790,rook-ceph-mon2=10.111.124.61:6790,rook-ceph-mon3=10.102.61.91:6790
2018-07-13 08:11:06.253766 I | op-mon: loaded: maxMonID=4, mons=map[rook-ceph-mon2:0xc4204280c0 rook-ceph-mon3:0xc420428100 rook-ceph-mon4:0xc420428080], mapping=&{Node:map[rook-ceph-mon2:0xc4202966c0 rook-ceph-mon3:0xc4202966f0 rook-ceph-mon4:0xc420296720] Port:map[]}
2018-07-13 08:11:06.254053 I | exec: Running command: rbd map rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864 --id admin --cluster=rook --keyring=/tmp/rook.keyring135704346 -m 10.106.8.20:6790,10.111.124.61:6790,10.102.61.91:6790 --conf=/dev/null
2018-07-13 08:12:06.254712 I | exec: Timeout waiting for process rbd to return. Sending interrupt signal to the process
2018-07-13 08:12:06.256807 E | flexdriver: Attach volume rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864 failed: failed to attach volume rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864: failed to map image rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864 cluster rook. failed to map image rook-replicapool/pvc-f313aa20-85d0-11e8-b7bc-7625bd182864: Failed to complete 'rbd': signal: interrupt. . output:
```
```
kubectl logs -n rook rook-ceph-osd-node1-nwlwg
.
.
.
018-07-13 07:55:20.914568 I | cephosd: creating and starting the osds
2018-07-13 07:55:20.914710 I | exec: Running command: lsblk /dev/sda --bytes --pairs --output NAME,SIZE,TYPE,PKNAME
2018-07-13 07:55:20.916799 I | exec: Running command: udevadm info --query=property /dev/sda2
2018-07-13 07:55:20.919077 I | exec: Running command: udevadm info --query=property /dev/sda2
2018-07-13 07:55:20.921580 I | exec: Running command: udevadm info --query=property /dev/sda3
2018-07-13 07:55:20.923485 I | exec: Running command: udevadm info --query=property /dev/sda3
2018-07-13 07:55:20.928374 I | exec: Running command: udevadm info --query=property /dev/sda1
2018-07-13 07:55:20.930382 I | exec: Running command: udevadm info --query=property /dev/sda1
2018-07-13 07:55:20.932383 I | exec: Running command: udevadm info --query=property /dev/sda
2018-07-13 07:55:20.934375 I | cephosd: skipping device sda that is in use (not by rook). fs: , ownPartitions: false
2018-07-13 07:55:20.934521 I | exec: Running command: lsblk /dev/vda --bytes --pairs --output NAME,SIZE,TYPE,PKNAME
2018-07-13 07:55:20.936605 I | exec: Running command: udevadm info --query=property /dev/vda14
2018-07-13 07:55:20.938452 I | exec: Running command: udevadm info --query=property /dev/vda14
2018-07-13 07:55:20.940367 I | exec: Running command: udevadm info --query=property /dev/vda15
2018-07-13 07:55:20.942233 I | exec: Running command: udevadm info --query=property /dev/vda15
2018-07-13 07:55:20.944288 I | exec: Running command: udevadm info --query=property /dev/vda1
2018-07-13 07:55:20.946136 I | exec: Running command: udevadm info --query=property /dev/vda1
2018-07-13 07:55:20.948052 I | exec: Running command: udevadm info --query=property /dev/vda
2018-07-13 07:55:20.949955 I | cephosd: skipping device vda that is in use (not by rook). fs: , ownPartitions: false
2018-07-13 07:55:20.958459 I | cephosd: configuring osd devices: {"Entries":{}}
2018-07-13 07:55:20.958630 I | cephosd: configuring removed osd devices: {"Entries":{}}
2018-07-13 07:55:20.958735 I | cephosd: configuring osd dirs: map[]
2018-07-13 07:55:20.958833 I | cephosd: configuring removed osd dirs: map[]
2018-07-13 07:55:20.960441 I | cephosd: removing osd devices: {"metadata":null,"entries":[]}
2018-07-13 07:55:20.960607 I | cephosd: removing osd dirs: map[]
2018-07-13 07:55:20.961410 I | cephosd: saving osd dir map
2018-07-13 07:55:20.967549 I | cephosd: sleeping a while to let the osds run...
2018-07-13 07:56:20.969736 I | exec: Running command: ceph osd dump --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/936998784
```

```
root@rook-tools:/# ceph -s
  cluster:
    id:     0fab1144-c234-4d11-9520-83a29660aabe
    health: HEALTH_WARN
            2 osds down
            2 hosts (2 osds) down
            1 root (2 osds) down
            Reduced data availability: 100 pgs inactive

  services:
    mon: 3 daemons, quorum rook-ceph-mon3,rook-ceph-mon4,rook-ceph-mon2
    mgr: rook-ceph-mgr0(active)
    osd: 2 osds: 0 up, 2 in

  data:
    pools:   1 pools, 100 pgs
    objects: 0 objects, 0 bytes
    usage:   0 kB used, 0 kB / 0 kB avail
    pgs:     100.000% pgs unknown
             100 unknown
```

```
root@rook-tools:/# ceph osd tree
ID CLASS WEIGHT  TYPE NAME                STATUS REWEIGHT PRI-AFF
-1       0.01639 root default
-2       0.00819     host node1
 0   hdd 0.00819         osd.0              down  1.00000 1.00000
-5       0.00819     host node2
 1   hdd 0.00819         osd.1              down  1.00000 1.00000
```
the cluster manifest:
```
apiVersion: v1
kind: Namespace
metadata:
  name: rook
---
apiVersion: rook.io/v1alpha1
kind: Cluster
metadata:
  name: rook
  namespace: rook
spec:
  dataDirHostPath: /var/lib/rook
  # cluster level storage configuration and selection
  storage:
    useAllNodes: false
    useAllDevices: false
    deviceFilter:
    metadataDevice:
    location:
    storeConfig:
      storeType: bluestore
      databaseSizeMB: 1024 # this value can be removed for environments with normal sized disks (100 GB or larger)
      journalSizeMB: 1024  # this value can be removed for environments with normal sized disks (20 GB or larger)
    nodes:
    - name: "node1"
      devices:             # specific devices to use for storage can be specified for each node
      - name: "sda"
    - name: "node2"
      devices:             # specific devices to use for storage can be specified for each node
      - name: "sda"
```

```
lsblk -o NAME,LABEL
NAME    LABEL
sda
|-sda1
|-sda2
`-sda3
vda
|-vda1  cloudimg-rootfs
|-vda14
`-vda15 UEFI
```

**Expected behavior:**
The OSD should recover seamlessly and should survive a kubelet restart.

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04 LTS
* Kernel (e.g. `uname -a`): Linux node1 4.15.0-24-generic #26-Ubuntu SMP Wed Jun 13 08:44:47 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration: Digital-Ocean droplet
* Rook version (use `rook version` inside of a Rook Pod): rook: v0.7.0-136.gd13bc83
* Kubernetes version (use `kubectl version`): v1.11.0
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)): HEALTH_WARN 2 osds down; 2 hosts (2 osds) down; 1 root (2 osds) down; Reduced data availability: 100 pgs inactive

**Comment by galexrt** [comment]
This needs to be resolved in `master` then. Because `master` will be the upcoming `v0.8`.

**Comment by travisn** [comment]
@ivanovaleksandar I believe this was fixed with #1801. The build you are using likely had this issue in it. Can you try the latest master build?

**Comment by galexrt** [comment]
Seems that this issue does not happen anymore (no recent Slack users with the problem and/or issues opened).
